docs/doc-versions.adoc

```diff
@@ -0,0 +1,8 @@
+[[doc-versions]]
+[appendix]
+== Other documentation versions
+* http://docs.janusgraph.org/0.3.0/index.html[Version 0.3.0]
```

Comment:
0.3.0 hasn't been released yet... also, will the docs versions only list past versions, or all future versions as well? How will we keep these up-to-date, given that we don't modify historical build artifacts (including docs)?

Should we have a separate page on docs.janusgraph.org that is not part of the docs build that lists all possible versions, and have all the docs link there, so that there's no need to rebuild prior versions and they stay intact?

Reply:
@mbrukman For 0.3.0 I figured this would get merged when we did the release, which I'm led to believe should be in the near future(hopefully). 

I have a [PR in the docs project](https://github.com/JanusGraph/docs.janusgraph.org/pull/27) where I checked out all the release tags and replaced the Other Versions link to the release page with a link to the versions page. That PR should get merged before this one. 

I'm proposing we only have the dropdown on the most current supported versions and all the finalized releases will just have a link to doc-versions.html. When new releases are added the only thing that would need to be updated is the static doc-versions.html file in each release folder. Since the file will be identical across the releases it represents a trivial update that could easily be scripted in CI.

Reply:
This means we have to update `doc-versions.html` in every doc page, of every version; however, each page is built differently (i.e., the CSS/XML transforms are custom to every version), so if we change the CSS for some version (say, 0.4.0) we can't reuse that HTML verbatim back in 0.1.0 because then it will look wrong.

The correct thing to do in that case would be to run N new docs builds, one in each documentation directory, to ensure that it picks up the correct CSS/XML relevant for that version.

I would like to propose that we don't give ourselves this much work, and do the following:

* "Other Doc Versions" always links to a static link: `/latest/doc-versions.html`
* we keep updating the `doc-versions.adoc` file to add all the new versions
* each doc build will only include _prior_ versions, but that's fine: when the link is clicked via the top, users will find the complete set of versions, but in the doc they will only see earlier versions
* thus, in the body of `doc-versions.adoc` it should also say something like:

  > This page only lists versions released prior to ${version}; see [complete list of docs versions](/latest/doc-versions.html) for later versions."

Thoughts?

Reply:
I like the idea of just linking to `/latest/doc-versions.html`  I just did a diff between versions and they are less congruent than I was expecting. I should be able to update this PR as well as JanusGraph/docs.janusgraph.org#27 tomorrow morning. 

Thanks for the feedback.

Reply:
requested changes have been made. I'll setup a github pages preview once JanusGraph/docs.janusgraph.org#27 merges